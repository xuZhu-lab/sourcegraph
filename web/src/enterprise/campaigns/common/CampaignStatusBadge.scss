.campaign-status-badge {
    font-size: unset; // override .badge's smaller font-size

    display: flex;
    align-items: center;
}
