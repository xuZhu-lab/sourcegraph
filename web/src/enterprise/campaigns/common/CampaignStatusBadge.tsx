import React from 'react'
import classNames from 'classnames'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { CampaignsIcon } from '../icons'

interface Props {
    campaign: Pick<GQL.ICampaign, 'closedAt'>

    className?: string
}

/**
 * A badge that conveys the status of a campaign.
 */
export const CampaignStatusBadge: React.FunctionComponent<Props> = ({ campaign, className }) => (
    <span
        className={classNames(
            'campaign-status-badge badge',
            className,
            !campaign.closedAt ? 'badge-success' : 'badge-danger'
        )}
    >
        <CampaignsIcon className="icon-inline mr-1" /> {!campaign.closedAt ? 'Open' : 'Closed'}
    </span>
)
