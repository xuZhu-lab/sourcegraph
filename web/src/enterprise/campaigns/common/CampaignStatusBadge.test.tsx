import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { CampaignStatusBadge } from './CampaignStatusBadge'

describe('CampaignStatusBadge', () => {
    test('open', () =>
        expect(createRenderer().render(<CampaignStatusBadge campaign={{ closedAt: null }} />)).toMatchSnapshot())

    test('closed', () =>
        expect(
            createRenderer().render(<CampaignStatusBadge campaign={{ closedAt: '2020-01-01' }} />)
        ).toMatchSnapshot())
})
