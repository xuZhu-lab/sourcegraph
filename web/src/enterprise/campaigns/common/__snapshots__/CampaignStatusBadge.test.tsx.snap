// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignStatusBadge closed 1`] = `
<span
  className="campaign-status-badge badge badge-danger"
>
  <Memo(ImageAutoAdjustIcon)
    className="icon-inline mr-1"
  />
   
  Closed
</span>
`;

exports[`CampaignStatusBadge open 1`] = `
<span
  className="campaign-status-badge badge badge-success"
>
  <Memo(ImageAutoAdjustIcon)
    className="icon-inline mr-1"
  />
   
  Open
</span>
`;
